torch/csrc/onnx/onnx.h

```diff
@@ -284,4 +277,28 @@ class GraphProto : public MicroProto<onnx_GraphProto> {
   }
 };
 
+class ModelProto : public MicroProto<onnx_ModelProto> {
+private:
+  std::string producer_name;
+  std::string producer_version;
+  std::string domain;
+  std::string doc_string;
+  // graph is a static field in nanopb struct.
+  GraphProto graph;
```

Comment:
There is a better way to handle this case, see `Note [Callback for nested messages]` in `torch/csrc/onnx/onnx.options`.